internal/handlers/common/update.go

```diff
@@ -899,10 +899,73 @@ func newUpdateError(code commonerrors.ErrorCode, msg, command string) error {
 	return commonerrors.NewWriteErrorMsg(code, msg)
 }
 
+// newPathValidator creates a root of pathValidator.
+// It shouldn't be used as a node of other pathValidator.
+func newPathValidator() *pathValidator {
+	return &pathValidator{
+		paths:    map[string]*pathValidator{},
+		rootNode: true,
+	}
+}
+
+// pathValidator stores validated paths and checks for any conflicts
+// between them. The conflict is when one path points to the part of
+// another path.
+//
+// Examples of conflicts:
+// `foo.bar, foo.bar`
+// `foo.bar, foo.bar.x`
+//
+// This is not a conflict:
+// `foo.bar.x, foo.bar.y`.
```

Comment:
FWIW, all those examples do not show a need for a tree structure – that's just string prefixes. Is it worth complicating our code there?

Reply:
Checking just for string prefixes might be simpler solution, although there might be some other issues to handle like the fact that `v` is prefix for `vfoo` (without the dot)

Reply:
Well, not literally, but by segments. Our paths already store segments in a slice.

Reply:
That creates O(n2) time complexity, but I guess we don't expect many paths so it doesn't need to scale. I'm going to try this out then

Reply:
Human cognition scales much worse :)